goreq_test.go

```diff
@@ -837,14 +837,43 @@ func TestRequest(t *testing.T) {
 				ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 					Expect(r.Header.Get("User-Agent")).ShouldNot(BeZero())
 					Expect(r.Host).Should(Equal("foobar.com"))
-
 					w.WriteHeader(200)
 				}))
 				defer ts.Close()
-
 				req := Request{Uri: ts.URL, Host: "foobar.com"}
 				res, err := req.Do()
 				Expect(err).ShouldNot(HaveOccurred())
+				Expect(res.StatusCode).Should(Equal(200))
+			})
+			g.It("Should be able to POST multipart forms", func() {
+				ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					Expect(r.Host).Should(Equal("foobar.com"))
+					Expect(r.Header.Get("Content-Type")).Should(Equal("multipart/mixed"))
+
+					b, _ := ioutil.ReadAll(r.Body)
```

Comment:
I guess the current pattern in good the only problem i think is that you might need multiple files at a go so something like this might be good
```go
type File struct {
   Name string
   FieldName string
   Source io.Reader
}

type MultipartForm struct {
	Params   map[string]string // or url.Values
    Files []*File
}
```
There was also a bug where it wasn't setting the content type and boundary.
```go
req.Header.Set("Content-Type", w.FormDataContentType())
```